diff --git a/profiles/dkan/modules/dkan/dkan_sitewide/dkan_sitewide.module b/profiles/dkan/modules/dkan/dkan_sitewide/dkan_sitewide.module
index 72f1d9f..20aec6d 100644
--- a/profiles/dkan/modules/dkan/dkan_sitewide/dkan_sitewide.module
+++ b/profiles/dkan/modules/dkan/dkan_sitewide/dkan_sitewide.module
@@ -18,10 +18,10 @@ function dkan_sitewide_user_menu_block() {
     if (module_exists('gravatar')) {
       $output = l('<img alt="my user picture" src="' . gravatar_get_gravatar($user->mail, array('size' => '25')) . '">' , 'user', array('html' => TRUE));
     }
-    $output .= '<span class="links">' . l($user->name, 'user') . l('Logout', 'user/logout', array('attributes' => array('class' => array('sub')))) . '</span>';
+    $output .= '<span class="links">' . l($user->name, 'user') . l(t('Logout'), 'user/logout', array('attributes' => array('class' => array('sub')))) . '</span>';
   }
   else {
-    $output .= '<span class="links">' . l('Login', 'user/login') . l('Register', 'user/register', array('attributes' => array('class' => array('sub')))) . '</span>';
+    $output .= '<span class="links">' . l(t('Login'), 'user/login') . l(t('Register'), 'user/register', array('attributes' => array('class' => array('sub')))) . '</span>';
   }
   return $output;
 }
