diff --git a/profiles/dkan/modules/contrib/ctools/plugins/access/entity_bundle.inc b/profiles/dkan/modules/contrib/ctools/plugins/access/entity_bundle.inc
index e07a048..1442d7f 100644
--- a/profiles/dkan/modules/contrib/ctools/plugins/access/entity_bundle.inc
+++ b/profiles/dkan/modules/contrib/ctools/plugins/access/entity_bundle.inc
@@ -35,7 +35,8 @@ function ctools_entity_bundle_ctools_access_get_children($plugin, $parent) {
     $plugin['keyword'] = $entity_type;
     $plugin['description'] = t('Control access by @entity entity bundle.', array('@entity' => $entity_type));
     $plugin['name'] = $parent . ':' . $entity_type;
-    $plugin['required context'] = new ctools_context_required(t(ucfirst($entity_type)), $entity_type);
+    $required_context_type = $entity_type == 'taxonomy_term' ? 'term' : $entity_type;
+    $plugin['required context'] = new ctools_context_required(t(ucfirst($entity_type)), $required_context_type);
     $plugins[$parent . ':' . $entity_type] = $plugin;
   }
 
