diff --git a/profiles/dkan/modules/contrib/ctools/page_manager/plugins/tasks/user_password.inc b/profiles/dkan/modules/contrib/ctools/page_manager/plugins/tasks/user_password.inc
index 8ea93bc..fe0f8d5 100644
--- a/profiles/dkan/modules/contrib/ctools/page_manager/plugins/tasks/user_password.inc
+++ b/profiles/dkan/modules/contrib/ctools/page_manager/plugins/tasks/user_password.inc
@@ -54,9 +54,9 @@ function page_manager_user_user_password() {
    // If we aren't admin but already logged on, go to the user page instead.
   global $user;
   $admin = user_access('administer users');
-  if (!$admin && $user->uid) {
-    drupal_goto('user/'. $user->uid);
-  }
+//  if (!$admin && $user->uid) {
+//    drupal_goto('user/'. $user->uid);
+//  }
   // Load my task plugin
   $task = page_manager_get_task('user_password');
 
